### PR TITLE
Update indexer version

### PR DESCRIPTION
**Pull request by miiu96**
## Reasoning behind the pull request
-  Changed the indexer version with one with some extra checks on the `tags` index 
## Proposed changes
-  PR from `mx-chain-es-indexer-go` module https://github.com/multiversx/mx-chain-es-indexer-go/pull/248

## Testing procedure
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -16,7 +16,7 @@ require (
 	github.com/multiversx/mx-chain-communication-go v1.0.6
 	github.com/multiversx/mx-chain-core-go v1.2.16
 	github.com/multiversx/mx-chain-crypto-go v1.2.8
-	github.com/multiversx/mx-chain-es-indexer-go v1.4.11
+	github.com/multiversx/mx-chain-es-indexer-go v1.4.12-0.20230913085433-b348fc9baeed
```
Comment (iulianpascalau) [inline go.mod:19]: proper release after testing

File: genesis/parsing/accountsParser.go [hunk 2]
```diff
@@ -392,6 +393,8 @@ func (ap *accountsParser) setScrsTxsPool(
 ) {
 	for _, id := range indexingData {
 		for txHash, tx := range id.ScrsTxs {
+			hexEncodedTxHash := hex.EncodeToString([]byte(txHash))
```
Comment (iulianpascalau) [inline genesis/parsing/accountsParser.go:396]: :+1:

File: go.mod (hunk 1 above)
Comment (miiu96) [reply go.mod]: done

**Review by mariusmihaic — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by liviuf-ancas — APPROVED** [review]
green on all system tests

**Pull request merged**